### ISSUE TITLE
[BUG] igraph demo notebook needs updating

### ISSUE THREAD

**Issue by lmeyerov** [graphistry/pygraphistry#261]
**Describe the bug**

Needs fixes, modern conventions, etc:
- fix colors: https://github.com/graphistry/pygraphistry/issues/259
- show login methods
- probably helps to show how to move to/from dataframes
- pointer to cuGraph for faster analytics

**Comment by iosonofabio** [comment]
hey, igraph developer here. Would be nice to have this demo working, anything we can do from our side?

Also, we are in the process of modernizing the igraph (python side) plotting and support Cairo, matplotlib, and plotly. Would be easy to incude this demo on our web page too to show compatibility.

**Comment by lmeyerov** [comment]
pygraphistry  has been evolving a bit, I'm thinking the 'better' fix would be cleaner bindings:
 
---

Dedicated ingest:

```python
# initial: convert G => pd.DataFrame 
# -- set col bindings: ._source, ._destination, ._node, ._edge
# -- by default, also cache source G reference as ._igraph
g = graphistry.from_igraph(G)
g = graphistry.nodes(G).edges(G) # builds upon existing ingest

# override
g2 = g.nodes(G) # set ._node if does not exist, else remap to existing g._node
g2 = g.edges(G) # set._node if does not exist, else remap to existing g._source/g._destination/g._edge

# encodings
# for the above, if G has any encodings, like size/color/label/..., optional flag "set_encodings=False" to pass them through
```

Dedicated export:

```python
G = g.to_igraph() # from ._edges df , and if exists, ._nodes df; same bindings as above
# note that we might have multiple engines like cudf / dask_cudf, but we can assume just pd.dataframe for now
```

---

A key missing use case is when using IG for enrichments, like computing page rank, where a flow might look like:

```python
g = graphistry.edges(pd.read_csv('http://some_data.csv'))
g2 = g.compute.pagerank(engine='igraph', normal_arg=1, extra=...).encode_point_color('pagerank', ['red', 'yellow'])
g2.plot()
```

or more composably:

```python
g = graphistry.edges(pd.read_csv(...))

G = g.to_igraph()... # run pagerank, ...
g2 = graphistry.nodes(G) # fresh graph

# new
g3 = g.merge_nodes(g2, merge='left' | 'right' | 'inner', conflict='left' | 'right' | prefix)
```

(we have a related merge discussion going for UMAP merging)

cc @silkspace 

thoughts? getting a good minimal flow up would be helpful, we've already had users asking about this

**Comment by lmeyerov** [comment]
@iosonofabio  FYI I started updating the graphistry pandas <> igraph interop handlers to roughly follow this : https://github.com/graphistry/pygraphistry/pull/342

**Comment by lmeyerov** [comment]
@iosonofabio  just published (`0.25.0`) w/ updates to the igraph bindings + 2 demo ipynb's

Adding something like `ig.plot(engine='graphistry')` or `ig.to_graphistry().plot()` would be super cool , and looks like a couple of tutorial pages (tutorial.html, visualization.html) to steer folks . What would be sensible steps for going about that, maybe starting by opening an issue in the github to propose it?

**Comment by lmeyerov** [comment]
(closing immediate issue as now works)

![Screenshot from 2022-05-01 01-29-54](https://user-images.githubusercontent.com/4249447/166138248-cd3a5949-9870-4edb-b951-874b4097bcd6.png)
